ui/src/app/services/sse.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable, NgZone } from '@angular/core';
+import {Observable} from "rxjs";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SseService {
+    private base = "/api/v1";
+    private evtSource: EventSource
+
+    constructor(private _zone: NgZone) {
+        this.evtSource = new EventSource(this.base + "/notifications");
+    }
+
+    subscribeTo(eventTypes: string[]): Observable<any> {
+        return new Observable( (subscriber) => {
+            eventTypes.forEach(eventType => {
+                this.evtSource.addEventListener(eventType,  (event) => {
+                    this._zone.run(() => {
```

Comment:
I tried on my side, it seems that running into _zone is not necessary: the event is propaagted and the new Devfile is correctly displayed in the UI

Reply:
Ah, interesting..
I took inspiration from various sources out there (like [here (in French)](https://blog.octo.com/angular-2-sse-et-la-detection-de-changements/),  [here](https://medium.com/@chrisbautistaaa/server-sent-events-in-angular-node-908830cc29aa), or [here](https://javascript.plainenglish.io/angular-and-server-sent-events-sse-54c29b4f174d)), and all of them relied on an Angular Zone. The previous version used `eventsource.onmessage`, but I changed it to `eventsource.addEventListener` when I wanted subscribers to be more specific on the event types they are interested in.
I'll remove it then if it works without this.
 Thanks.